src/software/thunderscope/robot_diagnostics/robot_view.py

```diff
@@ -130,9 +134,17 @@ def refresh(self) -> None:
         Until the buffer is empty
         """
         robot_status = self.robot_status_buffer.get(block=False, return_cached=False)
+        round_trip_time = self.round_trip_time_buffer.get(
+            block=False, return_cached=False
+        )
 
-        while robot_status is not None:
-            self.robot_view_widgets[robot_status.robot_id].update(robot_status)
+        while robot_status is not None and round_trip_time is not None:
+            self.robot_view_widgets[robot_status.robot_id].update(
+                robot_status, round_trip_time
+            )
             robot_status = self.robot_status_buffer.get(
                 block=False, return_cached=False
             )
+            round_trip_time = self.round_trip_time_buffer.get(
+                block=False, return_cached=False
+            )
```

Comment:
This while loop currently keeps updating the widgets for every robot_status-round_trip_time pair, and only the last pair ends up actually being shown on Thunderscope since all of the previous ones are instantly overwritten. So we could probably improve it by having the while loop go through the entirety of both buffers (probably better to do a separate while loop per buffer too so they both get emptied out even if somehow they are not the same size). Then `if robot_status is not None and round_trip_time is not None` we could `.update` the widgets.